fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadReportingServiceImpl.java

```diff
@@ -248,7 +250,8 @@ public String retrieveReportPDF(final String reportName, final String type, fina
 
             final Document document = new Document(PageSize.B0.rotate());
 
-            PdfWriter.getInstance(document, new FileOutputStream(new File(fileLocation + reportName + ".pdf")));
+            String validatedFileLocation = ESAPI.encoder().encodeForOS(new UnixCodec(), fileLocation + reportName + ".pdf");
```

Comment:
Have you been able to test this? I wanna make sure we are not breaking anything.